### DISCUSSION TITLE
Non-nested JSON data from `stdin` to CSV data in `stdout`/file using qsv only

### DISCUSSION THREAD

**Discussion by rzmk** [dathere/qsv#1878] [Q&A] [answered]
Nested JSON data may be complicated but my use case is for simpler data. For example say I'm working with `fruits.csv`:

```bash
qsv select 1- fruits.csv
```

```csv
fruit,price
apple,2.50
banana,3.00
strawberry,1.50
```

I get the stats data within my terminal:

```bash
qsv stats fruits.csv
```

```csv
field,type,is_ascii,sum,min,max,range,min_length,max_length,mean,stddev,variance,nullcount,max_precision,sparsity
fruit,String,true,,apple,strawberry,,5,10,,,,0,,0
price,Float,,7,1.5,3.0,1.5,4,4,2.3333,0.6236,0.3889,0,1,0
```

Then I convert that terminal output to JSON format with `qsv sqlp`:

```bash
qsv stats fruits.csv | qsv sqlp - 'SELECT * FROM _t_1' --format json
```

```json
[{"field":"fruit","type":"String","is_ascii":true,"sum":null,"min":"apple","max":"strawberry","range":null,"min_length":5,"max_length":10,"mean":null,"stddev":null,"variance":null,"nullcount":0,"max_precision":null,"sparsity":0},{"field":"price","type":"Float","is_ascii":null,"sum":7,"min":"1.5","max":"3.0","range":1.5,"min_length":4,"max_length":4,"mean":2.3333,"stddev":0.6236,"variance":0.3889,"nullcount":0,"max_precision":1,"sparsity":0}]
```

However I don't know of a reverse operation with qsv where JSON data is taken as input and output as CSV (similar to `qsv jsonl`).

My use case is converting this non-nested JSON data from `stdin` back into CSV format choosing between `stdout` and a file (`--output`) using qsv only (no jq, etc. due to platform considerations, added complexity, etc.). This is assuming I don't want to recompute a file with a qsv command to get it in CSV format and also this could be other kinds of JSON data that are not necessarily from qsv command output.

Is this possible and how?

**Comment by rzmk** [comment, marked as answer]
See PR #1880 for `qsv jsonp`.

**Reply by jqnatividad** [reply]
Awesome @rzmk !

You scratched your own itch and made a new command!
